apps/web/src/lib/variable-models/managed/asset-data-key-variable-model.ts

```diff
@@ -34,36 +34,23 @@ export default class AssetDataKeyVariableModel implements IBaseVariableModel {
 
             if (!this.#fetcher) this.#fetcher = getCancellableFetcher(SpaceConnector.clientV2.inventory.cloudServiceQuerySet.list);
 
-            const _query: Record<string, any> = {
-                only: ['data_keys'],
-                filter: [
-                    {
-                        key: 'data_keys',
-                        value: null,
-                        operator: 'not',
-                    },
-                ],
-            };
-            if (query.filters?.length) {
-                apiQueryHelper.setFilters([{ k: 'data_keys', v: query.filters, o: '=' }]);
-                _query.filter.push(...(apiQueryHelper.data?.filter ?? []));
-            }
-            if (query.search) {
-                _query.filter.push({
-                    key: 'data_keys',
-                    value: query.search,
-                    operator: 'contain',
-                });
-            }
+            apiQueryHelper.setOnly('data_keys')
+                .setFilters([{
+                    k: 'data_keys',
+                    v: null,
+                    o: '!=',
+                }]);
             const { status, response } = await this.#fetcher({
-                ...dependencyOptions, // TODO: check its working
-                query: _query,
+                ...dependencyOptions,
+                query: apiQueryHelper.data,
             });
             if (status === 'succeed' && response.results?.length) {
                 const target = response.results[0]?.data_keys ?? [];
-                this.#response = {
-                    results: target.map((d) => ({ key: d, name: d })),
-                };
+                let _results = target.map((d) => ({ key: d, name: d }));
+                if (query.search) {
+                    _results = _results.filter((d) => d.name.toLowerCase().includes(query.search?.toLowerCase() as string));
```

Comment:
suggestion: I think it would be better to use getHighlightText() like other autocomplete features!